h/static/scripts/helpers/tag-helpers.coffee

```diff
@@ -0,0 +1,58 @@
+createTagHelpers = ['localstorage', (localstorage) ->
+  TAGS_LIST_KEY = 'hypothesis.user.tags.list'
+  TAGS_MAP_KEY = 'hypothesis.user.tags.map'
+
+  filterTags: (query) ->
+    savedTags = localstorage.getObject TAGS_LIST_KEY
+    savedTags ?= []
+
+    # Only show tags having query as a substring
+    filterFn = (e) ->
+     e.toLowerCase().indexOf(query.toLowerCase()) > -1
+
+    savedTags.filter(filterFn)
+
+  # Add newly added tags from an annotation to the stored ones and refresh
+  # timestamp for every tags used.
+  refreshTags: (existingTags, allTags) ->
```

Comment:
I'm confused about this. It looks like you call it with the model value and the view value for the tags. That means that `allTags` is definitely not all of the tags in storage (`savedTags` is) and `existingTags` is... ?

Can you come up with some clearer names for these things? Perhaps the API is awkward, too. Why do we need both arguments?